File: cases/taint/csharp/argument-position-positive/ArgumentPositionPositive.cs
namespace DataFlowBench;

static class ArgumentPositionPositive
{
    static int dfb_source() // DFB-SOURCE: argument-position-input
    {
        return 1;
    }

    static int ChooseFirst(int first, int second) // DFB-WITNESS: argument-position-first
    {
        return first;
    }

    static void dfb_sink(int value) { } // DFB-SINK: argument-position-sink

    static void Run()
    {
        int result = ChooseFirst(dfb_source(), 0);
        dfb_sink(result);
    }
}

File: cases/taint/go/argument-position-positive/argument_position_positive.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: argument-position-input
	return 1
}

func chooseFirst(first int, second int) int { // DFB-WITNESS: argument-position-first
	return first
}

func dfb_sink(value int) {} // DFB-SINK: argument-position-sink

func run() {
	result := chooseFirst(dfb_source(), 0)
	dfb_sink(result)
}
